test/e2e/providers/aws/aws.go

```diff
@@ -135,6 +135,7 @@ func (a *Provider) GenerateMachineSet(withIgnoreLabel bool, replicas int32, wind
 		Placement:          linuxWorkerSpec.Placement,
 		UserDataSecret:     &core.LocalObjectReference{Name: clusterinfo.UserDataSecretName},
 	}
+	providerSpec.SetGroupVersionKind(mapi.SchemeGroupVersion.WithKind("AWSMachineProviderConfig"))
```

Comment:
Does this need to be a separate call? Can it be done as part of the struct construction?

Reply:
[the struct ](https://github.com/openshift/api/blob/f5b09d13c01f3a2a4228744f87fd84066adf10d9/machine/v1beta1/types_awsprovider.go#L12 )doesn't have a place where the groupversionkind can be set, because GroupVersionKind is a property of the [runtime package](https://pkg.go.dev/k8s.io/apimachinery/pkg/runtime#TypeMeta.SetGroupVersionKind), not the AWSMachineProviderConfig

Reply:
you can't just set .typemeta.Kind?

Reply:
ohh, I see what you mean! 👍
fixed!

Reply:
adding it to the typemeta was causing an error 

> error waiting for valid ConfigMap windows-services-10.19.0-dec881d-dirty: error waiting for ConfigMap openshift-windows-machine-config-operator/windows-services-10.19.0-dec881d-dirty: timed out waiting for the condition 

I think this is because you're supposed to use the setgroupversionkind when you're using a registered scheme, instead of using the TypeMeta directly, because it gets serialized wrong if it's not set through the function.